control-plane/subcommand/gateway-resources/command_test.go

```diff
@@ -254,3 +268,95 @@ func TestRun(t *testing.T) {
 		})
 	}
 }
+
+var meshGWConfigFileContent = `gatewayClassConfigs:
+  - apiVersion: mesh.consul.hashicorp.com/v2beta1
+    metadata:
+      name: consul-mesh-gateway
+      namespace: namespace
+    kind: gatewayClassConfig
+    spec:
+      deployment:
+meshGateways:
+  - name: mesh-gateway
+    spec:
+      gatewayClassName: consul-mesh-gateway
+`
+
+func TestRun_loadMeshGatewayConfigs(t *testing.T) {
+	filename := createMeshGatewayConfigFile(t)
+	// setup k8s client
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                            ui,
+		k8sClient:                     client,
+		flagMeshGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadMeshGatewayConfigs()
+	require.NoError(t, err)
+	require.NotEmpty(t, cmd.meshGatewayConfig.GatewayClassConfigs)
+
+	// we only created one class config
+	classConfig := cmd.meshGatewayConfig.GatewayClassConfigs[0].DeepCopy()
+
+	// TODO: Add resources to the example yaml and test here once https://github.com/hashicorp/consul/pull/19725 merges
```

Comment:
Going to update this test and the example yaml above once the PR linked in the comment merges so we can add the resources as well